core/editor/src/tools/select.rs

```diff
@@ -0,0 +1,40 @@
+use crate::events::Event;
+use crate::events::MouseKeys;
+use crate::tools::Tool;
+use document_core::Operation;
+
+#[derive(Default)]
+pub struct Select(Fsm);
+
+impl Tool for Select {
+	fn handle_input(&mut self, event: Event) -> Option<Operation> {
+		match event {
+			Event::MouseDown(state) => {
+				if !(state.mouse_keys & MouseKeys::LEFT).is_empty() {
+					self.0 = Fsm::LmbDown;
+				}
+			}
+			Event::MouseUp(state) => {
+				if self.0 == Fsm::LmbDown && state.mouse_keys == MouseKeys::LEFT {
```

Comment:
This can be left untouched if my second comment from just above is implemented. Otherwise replace this with `state.mouse_keys.contains(MouseKeys::LEFT)`